ShellPkg/Library/UefiShellDebug1CommandsLib/HexEdit/BufferImage.c

```diff
@@ -2069,16 +2064,12 @@ HBufferImageAddCharacterToBuffer (
   //
   // get a place to add
   //
-  for (Index = (INTN)(Size + Count - 1); Index >= (INTN)Pos; Index--) {

```

Comment:
Maybe a dumb question, but shouldn't this be instead:
```
CopyMem (&BufferPtr[Pos], &BufferPtr[Pos - Count], Size - Pos + Count);
```

Reply:
Sorry it took me a while to get back. Yes, I think you're right and I updated it.